web/profiles/joinup/src/Access/CanonicalRouteRestrict.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Restricts canonical routes to UID 1 only.
+ */
+class CanonicalRouteRestrict implements AccessInterface {
+
+  /**
+   * Constructs a CanonicalRouteRestrict object.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match object to be checked.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The account being checked.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(RouteMatchInterface $route_match, AccountInterface $account): AccessResultInterface {
+    $rdf_entity = $route_match->getParameter('rdf_entity');
+    if (!($rdf_entity instanceof RdfInterface) || $rdf_entity->bundle() !== 'spdx_licence') {
+      // In case the route the canonical route of the spdx_licence, return a
+      // positive access result as a neutral would deny the access to the route.
+      // @see \Drupal\Core\Access\AccessResult::andIf().
+      return AccessResult::allowed()->addCacheContexts(['user']);
+    }
+
+    if ($account->id() === 1) {
```

Comment:
I keep quoting myself (for how many times?):

> We should not perform access checks, neither of role name, nor on user ID. We should always check for permissions.

That said, we can add permission `bypass entity canonical route restriction` (this could be made more granular, on entity type, in the future) and do the check on `$account->hasPermission(...)`.

Note that the UID1 special behaviour will be removed in https://www.drupal.org/project/drupal/issues/540008

Reply:
I copied this over from our UID1Only access handler. I am going to remove it entirely. No user will have access to the canonical route. Or do you think there is any value for that?